nise/__main__.py

```diff
@@ -43,14 +59,49 @@ def create_parser():
                         required=True,
                         type=argparse.FileType('w'),
                         help='Generated output file')
+    parser.add_argument('--s3-bucket-name',
+                        metavar='BUCKET_NAME',
+                        dest='bucket_name',
+                        required=False,
+                        help='AWS S3 bucket to place the data.')
+    parser.add_argument('--s3-report-name',
+                        metavar='COST_REPORT_NAME',
+                        dest='report_name',
+                        required=False,
+                        help='Directory path to store data in the S3 bucket.')
     return parser
 
 
+def _check_s3_arguments(parser, options):
+    """Validate s3 argument combination.
+
+    Args:
+        parser (Object): ArgParser parser.
+        options (Dict): dictionary of arguments.
+    Raises:
+        (ParserError): If combination is invalid.
+
+    """
+    bucket_name = options.get('bucket_name')
+    report_name = options.get('report_name')
+    s3_valid = False
+    if bucket_name and report_name:
+        s3_valid = True
+    elif not bucket_name and not report_name:
+        s3_valid = True
+    if not s3_valid:
+        msg = 'Both {} and {} must be supplied, if one is provided.'
+        msg = msg.format('--s3-bucket-name', '--s3-report-name')
+        parser.error(msg)
```

Comment:
Will this kill the app if we get here?

Reply:
Yes, does a system exit with an output error.
```
$ nise --start-date 06-20-2018 --output-file test.csv --s3-bucket-name nisetestbucket
usage: nise [-h] --start-date DATE [--end-date DATE] --output-file FILE
            [--s3-bucket-name BUCKET_NAME] [--s3-report-name COST_REPORT_NAME]
nise: error: Both --s3-bucket-name and --s3-report-name must be supplied, if one is provided.
```